### PR TITLE
Fix S2Layer base class and props.

### PR DESCRIPTION
**Pull request by 1chandu**
Fix for #800

### CODE REVIEW

File: examples/layer-browser/src/examples/sample-layers.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ const S2LayerExample = {
     opacity: 0.6,
     getS2Token: f => f.token,
     getPath: f => f.path,
-    getFillColor: f => [f.value * 256, (1 - f.value) * 256, (1 - f.value) * 128],
+    getStrokeColor: f => [255, 255, 128],
```
Comment (1chandu) [inline examples/layer-browser/src/examples/sample-layers.js]: >     getStrokeColor: f => [255, 255, 128],
>     getFillColor: f => [255, 255, 128],

When color-picker.js calls these methods, it doesn't pass any parameters, `f` is going to be undefined, kept them in same style as `getStrokeWidth` below, but may be we should just change them to `callback: () => [255, 255, 128]` ?

File: examples/sample-layers/s2-layer/s2-layer.js [hunk 2]
```diff
@@ -64,13 +64,17 @@ const defaultProps = {
   getFillColor: f => f.fillColor || defaultFillColor
 };
 
-export default class S2Layer extends Layer {
+export default class S2Layer extends CompositeLayer {
+  constructor(props) {
+    super(props);
+  }
+
   initializeState() {
```
Comment (ibgreen) [inline examples/sample-layers/s2-layer/s2-layer.js]: I think this can be cut on compositeLayers.

Comment (1chandu) [inline examples/sample-layers/s2-layer/s2-layer.js]: good catch, I added it when debugging, forgot to remove.

Comment (ibgreen) [inline examples/sample-layers/s2-layer/s2-layer.js]: I was referring to  `initializeState` but I guess the constructor could also be cut.

File: examples/layer-browser/src/examples/sample-layers.js (hunk 1 above)
Comment (ibgreen) [inline examples/layer-browser/src/examples/sample-layers.js]: Not sure what happened to my comments, but we should keep the feature dependent fill color or we lose the ability to show neighbouring S2 cells, which is a key part of the demo.

We need to fix this in another place

Comment (1chandu) [inline examples/layer-browser/src/examples/sample-layers.js]: Fixed here : #870

**Pull request merged**